gemfirexd/client/src/main/java/io/snappydata/jdbc/ClientPoolDriver.java

```diff
@@ -0,0 +1,174 @@
+        // does.
+        if (ClientConfiguration.exceptionsOnLoadResources != null) {
+            exceptionsOnLoadDriver__ = Utils.accumulateSQLException(
+                    ClientConfiguration.exceptionsOnLoadResources,
+                    exceptionsOnLoadDriver__);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public boolean acceptsURL(String url) throws SQLException {
+        return (url != null && matchProtocol(url).matches());
+    }
+
+    @Override
+    protected boolean useThriftProtocol(Matcher m) {
+        String subProtocolGroup = m.group(2);
+        String protocol = m.group(1);
+        // if USE_THRIFT_AS_DEFAULT_PROP has been set explicitly then use that
+        // for default value else use the protocol string (jdbc:snappydata://
+        //   defaults to thrift while jdbc:gemfirexd:// defaults to old DRDA)
+        return subProtocolGroup == null || subProtocolGroup.length() == 0
+                ? ClientSharedUtils.isUsingThrift(
+                protocol.equalsIgnoreCase(SNAPPY_PROTOCOL))
+                : "thrift:pool:".equalsIgnoreCase(subProtocolGroup)
+                || "pool:".equalsIgnoreCase(subProtocolGroup);
+    }
+
+    @Override
+    protected Matcher matchURL(String url) {
+        return URL_PATTERN.matcher(url);
+    }
+
+    @Override
+    protected Matcher matchProtocol(String url) {
+        return PROTOCOL_PATTERN.matcher(url);
+    }
+
+    @Override
+    public Connection connect(String url, Properties properties) throws SQLException {
+
+        if (!acceptsURL(url)) {
+            System.out.println("Returning null connection From ClientPoolDriver ");
+            return null;
+        }
+
+        properties = (properties == null) ? getPoolProperties(url) : properties;
+        properties.setProperty(TomcatConnectionPool.POOL_URL, url);
+        properties.setProperty(TomcatConnectionPool.POOL_DRIVER_NAME, ClientPoolDriver.class.getName());
+
+        boolean creatingPool = Boolean.parseBoolean(properties.getProperty(POOL_INITIALIZATION));
+        if (creatingPool) {
+            return super.connect(url, properties);
+        }
+
+        TomcatConnectionPool p = poolMap.get(properties);
+        if (p != null) {
+            return p.getConnection();
+        } else {
+            TomcatConnectionPool pool = poolMap.get(properties);
+            if (pool == null) {
+                synchronized (ClientPoolDriver.class) {
+                    pool = poolMap.get(properties);
+                    if (pool == null) {
+                        pool = new TomcatConnectionPool(properties);
+                        poolMap.put(properties, pool);
+                    }
+                }
+            }
+            Connection conn = pool.getConnection();
+            return conn;
+        }
+    }
+
+    private Properties getPoolProperties(String url) {
+        Properties properties = new Properties();
```

Comment:
Just use empty. Let the pool use defaults. Check this. Let the pool handle the default case. Driver will be free from this.